route4me-csharp-sdk/Route4MeSDKLibrary/Route4MeManagerV5.cs

```diff
@@ -1757,56 +1931,534 @@ public Vehicle GetVehicleById(string vehicleId, out ResultResponse resultRespons
+        public Task<Tuple<Vehicle[], ResultResponse, string>> GetVehiclesByStateAsync(VehicleStates vehicleState)
+        {
+            var vehicleParams = new GenericParameters();
+            vehicleParams.ParametersCollection.Add("show", vehicleState.Description());
+
+
+            var result = GetJsonObjectFromAPIAsync<Vehicle[]>(
+                vehicleParams,
+                R4MEInfrastructureSettingsV5.Vehicles,
+                HttpMethodType.Get,
+                null,
+                false,
+                false);
+
+            return result;
+        }
+
+
+        #region Bulk Vehicle Operations
+
+        /// <summary>
+        /// Activates deactivated vehicles
+        /// </summary>
+        /// <param name="vehicleParams">The vehicle parameters</param>
+        /// <param name="resultResponse">Failing response</param>
+        /// <returns>Activated vehicles</returns>
+        public VehiclesResults ActivateVehicles(string[] vehicleIDs, out ResultResponse resultResponse)
+        {
+            var parameters = new GenericParameters();
+            foreach (var vehicleId in vehicleIDs)
+            {
+                parameters.ParametersCollection.Add("vehicle_ids[]", vehicleId);
+            }
+
+            var result = GetJsonObjectFromAPI<VehiclesResults>(
+                parameters,
+                R4MEInfrastructureSettingsV5.VehicleBulkActivate,
+                HttpMethodType.Post,
+                out resultResponse);
+
+            return result;
+        }
+
+        /// <summary>
+        /// Deactivates active vehicles
+        /// </summary>
+        /// <param name="vehicleParams">The vehicle parameters</param>
+        /// <param name="resultResponse">Failing response</param>
+        /// <returns>Deactivated vehicles</returns>
+        public Vehicle[] DeactivateVehicles(string[] vehicleIDs, out ResultResponse resultResponse)
+        {
+            var parameters = new GenericParameters();
+            foreach (var vehicleId in vehicleIDs)
+            {
+                parameters.ParametersCollection.Add("vehicle_ids[]", vehicleId);
+            }
+
+            var result = GetJsonObjectFromAPI<Vehicle[]>(
+                parameters,
+                R4MEInfrastructureSettingsV5.VehicleBulkDeactivate,
+                HttpMethodType.Post,
+                out resultResponse);
+
+            return result;
+        }
+
+        /// <summary>
+        /// Updates multiple vehicles at once asynchronously.
+        /// </summary>
+        /// <param name="vehicleArray">An object containing an array of the vehicles</param>
+        /// <returns>Returns a result as a tuple object:
+        /// - object: if success, this items is not null
+        /// - ResultResponse: if failed, this items is not null
+        /// - string: if success, this items is a job ID</returns>
+        public Task<Tuple<object, ResultResponse, string>> UpdateVehicles(Vehicles vehicleArray)
```

Comment:
Name-wise it is sync, but implementation-wise it is async. Please, align it. Rename to UpdateVehiclesAsync and introduct sync version with 'UpdateVehicles' name (like in another APIs).